client/src/extension.ts

```diff
@@ -171,6 +172,15 @@ function handleDocumentOpen(...documents: vscode.TextDocument[]) {
   }
 }
 
+function handleTextDocumentSave(doc: vscode.TextDocument) {
+  if (!LANGUAGES.includes(doc.languageId)) {
+    return;
+  }
+  if (extensionContext.workspaceSettings.cacheOnSave) {
+    vscode.commands.executeCommand("deno.cache");
```

Comment:
Is there a way to do this only if deno says there's stuff to cache in the current file? (ex. if there are diagnostics or cache quick fixes). This can be quite expensive.

Reply:
> Is there a way to do this only if deno says there's stuff to cache in the current file

I think so, will take a look soon...